### PR TITLE
DRAC-360 | fixed instructions details reversed order

### PR DESCRIPTION
**Pull request by stasmenklab**

### CODE REVIEW

File: src/modules/instruction/instruction.service.ts [hunk 1]
```diff
@@ -21,6 +21,10 @@ export class InstructionService {
   ) {}
 
   public async getAll(programId: number): Promise<InstructionDto[]> {
+    if(!programId){
```
Comment (Zhuravlev1) [inline src/modules/instruction/instruction.service.ts]: How is it possible? programId is not optional

File: src/modules/instruction/instruction.service.ts [hunk 2]
```diff
@@ -108,7 +112,7 @@ export class InstructionService {
       const camelCaseName = this.toCamelCase(instruction.name)
       const func = this.generateFunction(instruction.name, camelCaseName)
       code.push(...func)
-      const structure = this.generateStructure(instruction.elements, camelCaseName)
+      const structure = this.generateStructure(instruction.elements.reverse(), camelCaseName)
```
Comment (Zhuravlev1) [inline src/modules/instruction/instruction.service.ts]: it's better to reverse it on the DB level

**Review by Zhuravlev1 — CHANGES_REQUESTED** [review]

File: src/modules/instruction/instruction.service.ts (hunk 1 above)
Comment (stasmenklab) [reply src/modules/instruction/instruction.service.ts]: If I don't pass in the programId in as the query param in the url while making request, I can log in here as undefined

Comment (Zhuravlev1) [reply src/modules/instruction/instruction.service.ts]: but you can check it by JOI in the controller level, is not?

**Review by Zhuravlev1 — APPROVED** [review]

**Pull request merged**